examples/pytorch_distributed_BERT_training.py

```diff
@@ -0,0 +1,329 @@
+        self.bert_model = transformers.BertModel.from_pretrained("bert-base-uncased")
+        self.out = nn.Linear(768, 1)
+
+    def forward(self, ids, mask, token_type_ids):
+        _, o2 = self.bert_model(
+            ids, attention_mask=mask, token_type_ids=token_type_ids, return_dict=False
+        )
+
+        out = self.out(o2)
+
+        return out
+
+
+def finetune(epochs, dataloader, model, loss_fn, optimizer, rank, cpu_id):
+
+    ## initialize an output file to log behavior for a specific cpu
+    # output_filename = f"output_rank_{cpu_id}.txt"
+    # with open(output_filename, 'w') as file:
+    #     file.write(f"Output content for rank {cpu_id}\n")
+
+    model.train()
+
+    for epoch in range(epochs):
+        if isinstance(dataloader.sampler, DistributedSampler):
+            dataloader.sampler.set_epoch(epoch)
+
+        loop = tqdm(enumerate(dataloader), leave=False, total=len(dataloader))
+        for batch, dl in loop:
+
+            print(f"rank {cpu_id} epoch {epoch} batch {batch}")
+            # append_to_output(output_filename, f"rank {cpu_id} epoch {epoch} batch {batch}")
+
+            # below part can be commented out when testing data loading only
+            # start:
+            ids = dl["ids"].to(rank)
+            token_type_ids = dl["token_type_ids"].to(rank)
+            mask = dl["mask"].to(rank)
+            label = dl["target"].to(rank)
+            label = label.unsqueeze(1)
+
+            optimizer.zero_grad()
+
+            output = model(ids=ids, mask=mask, token_type_ids=token_type_ids)
+            label = label.type_as(output)
+
+            loss = loss_fn(output, label)
+            loss.backward()
+
+            optimizer.step()
+
+            pred = np.where(output.cpu().detach().numpy() >= 0, 1, 0)
+            label = label.cpu().detach().numpy()
+
+            num_correct = sum(1 for a, b in zip(pred, label) if a[0] == b[0])
+            num_samples = pred.shape[0]
+            accuracy = num_correct / num_samples
+
+            print(
+                f"Got {num_correct} / {num_samples} with accuracy {float(num_correct) / float(num_samples) * 100:.2f}"
+            )
+
+            loop.set_description(f"Epoch={epoch + 1}/{epochs}")
+            loop.set_postfix(loss=loss.item(), acc=accuracy)
+            # end
+    return model
+
+
+def preprocess(directory_path, chunk_size):
+    """
+    Preprocess each file in the directory for Dataset class.
+    :param directory_path: directory path contain the files to be processed
+    :param chunk_size: read file that has chunk_size to the memory each time
+    :return: processed_file_info is a dictionary that contain start_line number for each file;
+            total length of all files in the directory
+    """
+    fsspec.register_implementation("alluxiofs", AlluxioFileSystem, clobber=True)
+    alluxio_fs = fsspec.filesystem(
+        "alluxiofs", etcd_hosts="localhost", etcd_port=2379, target_protocol="s3"
+    )
```

Comment:
so in here, we are already in a child process right?
we can reuse the alluxio_fs instance created per child process when we spawn main() correct?